src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.cs

```diff
@@ -54,7 +55,10 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 return;
             }
```

Comment:
what aobut events?  can they not be 'new'?

Reply:
what about indexers?

Reply:
Nested types can also be `new`.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#the-new-modifier

Reply:
I think `new` can be used on every kind of member except constructors, finalizers, and operators.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEA3GUCWAZgJ4A+AAgEwCMAsAFANUAEAwgwN4PM/PkDMzAHYwA7nxTMAsgAoAlMw7MAvt14DhY5viEZmAMQDcanhpHiYuXX2qVmAUWP1efQee3WACouYBzGBiGKiaumuI6ehgAFvgAzgDaEcwADgC6zAC8AHwpTi7uLKx2SqrOvO6xGFAArmB6AMrFwWU87hF4hACGYDDMAJJNpflaACYwADYwvp0YveSSACKU8nnlWjBC1QC2DoMMykA

Reply:
I think I can use `token.GetAncestor<MemberDeclarationSyntax>();` instead of these multiple checks. However it will also affect CS0108 fix. That's not an issue if I need to move the entire fixer to a separate class/folder, as it was mentioned in another comment.

Reply:
> However it will also affect CS0108 fix. That's not an issue if I need to move the entire fixer to a separate class/folder, as it was mentioned in another comment.

i would just do this.

Reply:
Moved to a separate class and used `.GetAncestor<MemberDeclarationSyntax>()` there. Also tests are added for all cases from SharpLab example.